src/frontend/admin/src/components/presentational/discount/DiscountSelect.tsx

```diff
@@ -0,0 +1,262 @@
+      try {
+        const response = await DiscountRepository.getAll();
+        setDiscounts(response.results);
+      } catch (fetchError) {
+        setFormError(`Failed to fetch discounts: ${fetchError}`);
+      } finally {
+        setLoading(false);
+      }
+    };
+
+    fetchDiscounts();
+  }, []);
+
+  const handleOpenDialog = () => {
+    setDialogOpen(true);
+  };
+
+  const handleCloseDialog = () => {
+    setDialogOpen(false);
+    setNewDiscount({ amount: null, rate: null });
+    setFormError(null);
+  };
+
+  const handleCreateDiscount = async () => {
+    // Validate that either rate or amount is set, but not both
+    if (
+      (newDiscount.rate === null && newDiscount.amount === null) ||
+      (newDiscount.rate !== null && newDiscount.amount !== null)
+    ) {
+      setFormError(intl.formatMessage(messages.discountExclusiveHelperText));
+      return;
+    }
+
+    try {
+      if (newDiscount.rate !== null) {
+        newDiscount.rate /= 100; // Convert to decimal
+      }
+      const createdDiscount = await DiscountRepository.create(newDiscount);
+      setDiscounts([...discounts, createdDiscount]);
+      onChange(createdDiscount.id);
+      handleCloseDialog();
+    } catch (fetchError) {
+      setFormError(`Failed to create discount: ${fetchError}`);
+    }
+  };
+
+  const handleAmountChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const inputValue =
+      e.target.value === "" ? null : parseInt(e.target.value, 10);
+    setNewDiscount({ ...newDiscount, amount: inputValue, rate: null });
+  };
+
+  const handleRateChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const inputValue =
+      e.target.value === "" ? null : parseFloat(e.target.value);
+    setNewDiscount({ ...newDiscount, rate: inputValue, amount: null });
+  };
+
+  return (
+    <>
+      <FormControl fullWidth error={error}>
+        <InputLabel id="discount-select-label">
+          {intl.formatMessage(messages.discountSelectLabel)}
+        </InputLabel>
+        <Select
+          labelId="discount-select-label"
+          value={value || ""}
+          onChange={(e) =>
+            onChange(e.target.value === "" ? null : e.target.value)
+          }
+          label={intl.formatMessage(messages.discountSelectLabel)}
+          disabled={loading}
+        >
+          <MenuItem value="">
+            {intl.formatMessage(messages.noDiscountOption)}
+          </MenuItem>
+          {discounts.map((discount) => (
+            <MenuItem key={discount.id} value={discount.id}>
+              {getDiscountLabel(discount)}
+            </MenuItem>
+          ))}
+        </Select>
+        <FormHelperText>
+          {helperText || intl.formatMessage(messages.discountHelperText)}
+        </FormHelperText>
+        <Button
+          startIcon={<AddIcon />}
+          onClick={handleOpenDialog}
+          sx={{ mt: 1 }}
+        >
+          {intl.formatMessage(messages.createNewDiscountButton)}
+        </Button>
```

Comment:
Instead of display a button here, I`ll just put `+` in the right action slot of the select. Like it is done in SkillSelect for example.

<img width="278" alt="image" src="https://github.com/user-attachments/assets/08ba8d11-ea9b-4be5-a324-cdbb9be874b3" />